### PR TITLE
Allatori demo

### PR DESCRIPTION
**Pull request by enovella**
More info at https://github.com/rednaga/APKiD/issues/86

### CODE REVIEW

File: apkid/rules/dex/obfuscators.yara [hunk 1]
```diff
@@ -169,4 +169,17 @@ rule arxan_multidex : obfuscator
...
 
+rule allatori_demo : obfuscator
+{
+  meta:
+    description = "Allatori demo"
+    url         = "http://www.allatori.com/features.html"
+    author      = "Eduardo Novella"
+
+  strings:
+    $s = "ALLATORIxDEMO"
```
Comment (CalebFenton) [inline apkid/rules/dex/obfuscators.yara]: Is there any way to make this more specific? In the code you posted, it looked like it was used for method names, so maybe the string could be `;->ALLATORIxDEMO(`

Are there any other things this obfuscator does that we can capture with a rule here? Maybe some static decryption method or hard coded values? Any added files to the APK?

File: apkid/rules/dex/obfuscators.yara [hunk 2]
```diff
@@ -169,4 +169,17 @@ rule arxan_multidex : obfuscator
     not arxan
 }
 
+rule allatori_demo : obfuscator
+{
+  meta:
+    description = "Allatori demo"
+    url         = "http://www.allatori.com/features.html"
+    author      = "Eduardo Novella"
```
Comment (CalebFenton) [inline apkid/rules/dex/obfuscators.yara]: Could you add a sample hash?

File: apkid/rules/dex/obfuscators.yara (hunk 1 above)
Comment (enovella) [reply apkid/rules/dex/obfuscators.yara]: OK, fair enough. Although I have not tested, it is already pushed. 

I do not know anything else now. No assets files added to the APK that I am aware of.

File: apkid/rules/dex/obfuscators.yara (hunk 2 above)
Comment (enovella) [reply apkid/rules/dex/obfuscators.yara]: Done!

File: apkid/rules/dex/obfuscators.yara (hunk 1 above)
Comment (enovella) [reply apkid/rules/dex/obfuscators.yara]: The string ` ;->ALLATORIxDEMO(` does not match the rule @CalebFenton. Perhaps we can add the null terminator to increase the length.

File: apkid/rules/dex/obfuscators.yara [hunk 3]
```diff
@@ -169,4 +171,18 @@ rule arxan_multidex : obfuscator
...
+    example     = "7f2f5aac9833f7bdccc0b9865f5cc2a9c94ee795a285ef2fa6ff83a34c91827f"
+
+  strings:
+    $s = "ALLATORIxDEMO"
```
Comment (CalebFenton) [inline apkid/rules/dex/obfuscators.yara]: I thought I commented on this yesterday, but I must have forgot to submit?

This string is really naked. I saw the example code you posted was using this identifier as a method name, so is it possible to make this rule something like `;->ALLATORIxDEMO(`.

Also, are there any other modifications done that we could fingerprint? E.g. string decryption constants, opcodes, etc.

File: apkid/rules/dex/obfuscators.yara (hunk 2 above)
Comment (enovella) [reply apkid/rules/dex/obfuscators.yara]: Ready to be merged @CalebFenton

**Pull request merged**